app/src/main/java/com/isc/hermes/controller/GenerateRandomIncidentController.java

```diff
@@ -24,18 +33,55 @@ public GenerateRandomIncidentController(AppCompatActivity activity){
         generateRandomIncidentVIew = new GenerateRandomIncidentView(activity);
         Button generateButton = activity.findViewById(R.id.startGenerateIncidentButton);
         generateButton.setOnClickListener(v -> initGeneration());
+        generatorManager = new GeneratorManager();
+        incidentsUploader = IncidentsUploader.getInstance();
     }
 
     /**
      * This method initiates the incident generation process.
+     * <p>
+     * Gets the current location and input parameters received from the UI to generate the
+     * incidents, uses the generatorManager to generate incidents and builds a string with
+     * the structure of a json to upload the data to the database.
+     * </p>
      */
     private void initGeneration(){
```

Comment:
This method starts the generator and updates the incidents, this breaks the Single Responsibility principle, it must be refactored